### PR TITLE
users: New default config/data locations

### PR DESCRIPTION
**Pull request by calmh**
See syncthing/syncthing#9180

### CODE REVIEW

File: users/config.rst [hunk 1]
```diff
@@ -25,11 +26,12 @@ Description
 
 Syncthing uses a single directory to store configuration and crypto keys.
 Syncthing also has a database, which is often stored in this directory too.
-The config location defaults to ``$HOME/.config/syncthing``
-(Unix-like), ``$HOME/Library/Application Support/Syncthing`` (Mac),
-or ``%LOCALAPPDATA%\Syncthing`` (Windows). It can be changed at runtime
-using the ``--config`` flag. In this directory the following files are
-located:
+The config location defaults to ``$XDG_STATE_HOME/syncthing`` or
```
Comment (acolomb) [inline users/config.rst:29]:
```suggestion
The location defaults to ``$XDG_STATE_HOME/syncthing`` or
```

I think talking especially about a "config location" here can be confusing after the previous statement.  Please reflow at will if accepting the suggestion. :wink:

**Review by imsodin — APPROVED** [review]

File: users/config.rst [hunk 2]
```diff
@@ -48,22 +50,27 @@ located:
 
 The database is stored either in the same directory as the config (usually the
 default), but may also be located in one of the following directories (Unix-like
-platforms only):
+platforms only) for legacy reasons:
```
Comment (acolomb) [inline users/config.rst:53]: We should mention the legacy *config* location as well, i.e. `%XDG_CONFIG_HOME/syncthing` and `$HOME/.config/syncthing`, as they are is still tried.

File: users/config.rst [hunk 3]
```diff
@@ -48,22 +50,27 @@ located:
...
-The location of the database can be changed using the ``--data`` flag. The
-``--home`` flag sets both config and database locations at the same time.
-The database contains the following files:
+The database directory contains the following files, among others:
 
 :file:`index-{*}.db`
     A directory holding the database with metadata and hashes of the files
     currently on disk and available from peers.
 
+:file:`syncthing.log`
```
Comment (acolomb) [inline users/config.rst:66]: `csrftokens.txt` also lands here if I read the code correctly.  Move from the paragraph above please.

File: users/faq-parts/troubleshooting.rst [hunk 4]
```diff
@@ -128,7 +128,7 @@ The web GUI contains a ``Recent Changes`` button under the device list which
 displays changes since the last (re)start of Syncthing. With the ``--audit``
 option you can enable a persistent, detailed log of changes and most
 activities, which contains a ``JSON`` formatted  sequence of events in the
```
Comment (acolomb) [inline users/faq-parts/troubleshooting.rst:130]: Nitpick: Double space.  Which is fine after a full-stop (end of sentence), but not within.

```suggestion
activities, which contains a ``JSON`` formatted sequence of events in the
```

**Review by acolomb — CHANGES_REQUESTED** [review]
Nicely done, but some corrections necessary.

**Comment by acolomb** [comment]
Oh and we should maybe have a `..versionadded::` tag here somewhere, to make it explicit when the defaults changed.

**Comment by calmh** [comment]
I rewrote stuff to be, in my mind, a bit clearer. A new `versionadded` describes the change. Then we don't need to deal with the change/legacy further down in the doc. I deliberately left out mentioning csrftokens.txt (but missed removing it from the old config section) because it's a tiny implementation wart that should go away (live in the database, if anywhere) -- it's covered by the new "among others" on the database dir. :)

**Comment by calmh** [comment]
n.b. the `versionchanged` is next+1 since I think this is too dangerous a change to slide into an `rc.2`. We should delay merge accordingly (of the docs change), until after 1.26 release.

**Pull request merged**